connectors/protocol/connectors.py

```diff
@@ -179,18 +183,23 @@ async def connector_put(
             is_native=is_native,
         )
 
-    async def connector_exists(self, connector_id):
+    async def connector_exists(self, connector_id, include_deleted=False):
         try:
-            doc = await self.fetch_by_id(connector_id)
+            doc = await self.get_connector(connector_id, include_deleted)
             return doc is not None
-        except DocumentNotFoundError:
+        except ElasticNotFoundError:
             return False
         except Exception as e:
             logger.error(
                 f"Error while checking existence of connector '{connector_id}': {e}"
             )
             raise e
 
+    async def get_connector(self, connector_id, include_deleted=False):
+        return await self.api.connector_get(
+            connector_id=connector_id, include_deleted=include_deleted
+        )
+
```

Comment:
This method should be inlined

Reply:
For now it can be inlined 👍 at some point we will need to revisit this to be able to do `connector_index.get` (same as `connector_index.fetch_by_id` to replace fetch_by_id with get API call